ui/src/table.tsx

```diff
@@ -393,7 +393,7 @@ const
           /> : null
       }, [onFilterChange, setFiltersInBulk]),
       onColumnContextMenu = React.useCallback((col: WaveColumn, e: React.MouseEvent<HTMLElement>) => {
-        const menuFilters = col.filters || items.map(i => i[col.fieldName || col.key])
+        const menuFilters = (col.filters || items.map(i => i[col.fieldName || col.key])).filter(item => item !== '')
```

Comment:
Another loop through items seems unnecessary since they will contain a lot of duplicates anyway. The filter should be applied only once the duplicates are taken care of.